File: Data Structures & Algorithms/sqrtx/submission-3.java
class Solution {
    public int mySqrt(long x) {
        long l=0, r=x;
        long ans = x;
        while(l<=r){
            long m =(r-l)/2+l;
            if(m*m==x) 
                return (int)m;
            else if(m*m>x) {
                r=m-1;
            }
            else {
                ans = m;
                l=m+1;
            }
        }
        return (int)ans;
    }
}

File: Data Structures & Algorithms/string-encode-and-decode/submission-3.cpp
class Solution {
public:

    string encode(vector<string>& strs) {
        string ans = "";
        for(string s : strs){
            ans += s;
            ans += "§";
        }

        return ans;
    }

    vector<string> decode(string s) {
        string delimiter = "§";
        vector <string> ans ;
        int n = s.size();
        int last =0;
        for(int i=0;i<n;i++){
            if(s.compare(i, delimiter.size(), delimiter) == 0){
                ans.push_back(s.substr(last , i-last));
                last = i+2;
            }
        }

        return ans;
    }
};
